[OLXBridge] new bridge

wrobelda opened the pull request:
Adds new bridge for the OLX classifieds platform. Should support all international favors of it.

Based on https://github.com/MK-PL/OLX-RSS by Maciej Kawa.

wrobelda committed:
[OLXBridge] new bridge

wrobelda committed:
[OLXBridge] option to limit to shipping offers only

wrobelda committed:
[OLXBridge] set the feed title according to search query

dvikan commented:
Works but takes like 20s. Also lots of php notices.

wrobelda commented:
> Works but takes like 20s. 

Only the first run, though, subsequent runs will be much faster with `getSimpleHTMLDOMCached` hitting the cache. 

> Also lots of php notices.

Let me look into that.

wrobelda committed:
[OLXBridge] Fix PHP notices

wrobelda commented:
> Also lots of php notices.

Now fixed, thanks.

Bockiii commented:
Hi,

I just tested the other languages that OLX provides and none of them work. 

https://www.olx.ro/piese-auto/mecanica-electrica/q-toyota/
https://www.olx.com.br/brasil?q=toyota
https://www.olx.co.id/items/q-toyota
https://www.olx.pt/ads/q-toyota/
https://www.olx.in/items/q-toyota?isSearchCall=true

The only one that works is the polish version. Also, most of those sites look very different, so the css cant work.

wrobelda commented:
I tested the Romanian version during the development and it did absolutely work just fine... Let me check again.

wrobelda committed:
[OLXBridge] Remove trailing slash from the URL

wrobelda commented:
The problem with your European OLX examples is the trailing slash — updated the code to remove it. The other OLX websites are indeed incompatible, I will therefore explicitly whitelist those that are. Thanks for testing!

Bockiii commented:
@wrobelda  so?

wrobelda commented:
I need to add some filter on domain part of the URL. Any existing bridges that do something similar?

Bockiii commented:
You can use input patterns that only allow your chosen .tld

https://github.com/RSS-Bridge/rss-bridge/blob/master/bridges/GoodreadsBridge.php#L21

wrobelda committed:
[OLXBridge] filter out the imposed additional search categories

wrobelda committed:
[OLXBridge] limit search to 'new' OLX platform variants

wrobelda commented:
@Bockiii this is now ready, please review again. 

EDIT: actually, I have some improvement in works, I'll ping back when done.

wrobelda committed:
[OLXBridge] Parse date, add ID, description

Deep-crawl all results. Penalty is low, as we were doing this for almost all of the results, anyway, yet it allows to obtain a unique ID, an uncomplicated Date string and a description.

Requires ext-intl for parsing the date according to locale.

wrobelda committed:
[OLXBridge] Parse date, add ID, description

Deep-crawl all results. Penalty is low, as we were doing this for almost all of the results, anyway, yet it allows to obtain a unique ID, an uncomplicated Date string and a description.

Requires ext-intl for parsing the date according to locale.

wrobelda committed:
[OLXBridge] Images are optional, handle appropriately

wrobelda committed:
[OLXBridge] handle the ID coming from sibling auto-moto portal

wrobelda committed:
[OLXBridge] handle the photos coming from sibling auto-moto portal

wrobelda committed:
[OLXBridge] use meta property to find img URL

wrobelda committed:
[OLXBridge] handle the date coming from sibling auto-moto portal

wrobelda committed:
[OLXBridge] use simplified syntax to retrieve content attribute value

wrobelda committed:
[OLXBridge] handle the description coming from sibling auto-moto portal

wrobelda committed:
[OLXBridge] fix phpcs complaints

wrobelda committed:
[OLXBridge] add categories

wrobelda committed:
[OLXBridge] handle the categories coming from sibling auto-moto portal

wrobelda committed:
[OLXBridge] hint image MIME type

OLX images have no obvious extension

wrobelda committed:
[OLXBridge] Fix content formatting

wrobelda commented:
@Bockiii should be ready now

wrobelda commented:
@dvikan this is ready now, notices should not happen anymore

wrobelda committed:
[OLXBridge] URL is pattern-checked, so no need to check again

wrobelda committed:
[OLXBridge] return actual search query as URI

dvikan commented on composer.json:
```diff
@@ -28,7 +28,8 @@
         "ext-simplexml": "*",
-        "ext-json": "*"
+        "ext-json": "*",
+        "ext-intl": "*"
```
Place this in the `"suggest"` section instead.

wrobelda replied:
This is needed for `IntlDateFormatter`, which, in turn, is needed to parse the localized date formats. For any platform that supports multiple languages, this handles all their local nuances automatically, so I'd argue that this is a valuable addition and a fairly common use-case for scraping, and sooner or later another Bridge would likely need it anyway.

wrobelda replied:
Ping @dvikan

dvikan commented on Dockerfile:
```diff
@@ -9,10 +9,12 @@ RUN apt-get update && \
       libzip-dev \
-      libmemcached-dev && \
+      libmemcached-dev \
+      libicu-dev && \
```
Is this `libicu-dev` really needed? Just making sure. Don't like bloating the deps.

wrobelda replied:
Required by `ext-intl`, as noted above.

dvikan commented:
This branch has conflicts that must be resolved

wrobelda commented:
> This branch has conflicts that must be resolved

Done
